src/Workspaces/CSharp/Portable/Simplification/CSharpSimplificationService.Expander.cs

```diff
@@ -271,6 +295,111 @@ public override SyntaxNode VisitArgument(ArgumentSyntax node)
                 return newArgument;
             }
 
+            private static bool CanMakeNameExplicitInTuple(TupleExpressionSyntax tuple, string name)
+            {
+                if (name == null || IsTupleElementNameReserved(name) != -1)
+                {
+                    return false;
+                }
+
+                bool found = false;
+                foreach (var argument in tuple.Arguments)
+                {
+                    string elementName = null;
+                    if (argument.NameColon != null)
+                    {
+                        elementName = argument.NameColon.Name.Identifier.ValueText;
+                    }
+                    else
+                    {
+                        elementName = ExtractAnonymousTypeMemberName(argument.Expression).ValueText;
+                    }
+
+                    if (elementName?.Equals(name, StringComparison.Ordinal) == true)
+                    {
+                        if (found)
+                        {
+                            // No duplicate names allowed
+                            return false;
+                        }
+                        found = true;
+                    }
+
+                }
+
+                return true;
+            }
+
+            // This code is roughly copied from the compiler
+            private static int IsTupleElementNameReserved(string elementName)
```

Comment:
Is this something that should be moved to one of the "Facts" classes in the compiler public API?